### PR TITLE
Add Docker image support multi OS archs

### PR DESCRIPTION
**Pull request by wilsonwu**
**What type of PR is this?**
Update Dockerfile to support different OS archs support, and update related script.
/kind feature

**What this PR does / why we need it**:
More requirements for ARM support.

### CODE REVIEW

**Review by shaneutt — COMMENTED** [review]
/ok-to-test

What version of `docker` did you use to test this?

**Comment by wilsonwu** [comment]
> /ok-to-test
> 
> What version of `docker` did you use to test this?

Tested in 20.10.11

**Comment by shaneutt** [comment]
I'm a little confused how this compares/contrasts with https://github.com/kubernetes-sigs/gateway-api/pull/1402. Have you see this one? It would seem that with #1402 we would simply have a multi-arch image rather than having the ability to build a single image for separate architectures. What are your thoughts?

**Comment by wilsonwu** [comment]
> I'm a little confused how this compares/contrasts with #1402. Have you see this one? It would seem that with #1402 we would simply have a multi-arch image rather than having the ability to build a single image for separate architectures. What are your thoughts?

#1402 is same as this PR, I think the #1402 is OK for support multi OS Archs, let close this after the #1402 done.

And use buildx to build images by manifest is better way.

**Comment by robscott** [comment]
Hey @wilsonwu, since #1402 has gone stale, I think we can use yours. Do you mind updating this PR to include `hack/build-and-push.sh` updates as well?

File: README.md [hunk 1]
```diff
@@ -68,3 +68,10 @@ Participation in the Kubernetes community is governed by the
 [concepts]: https://gateway-api.sigs.k8s.io/concepts/api-overview
 [security-model]: https://gateway-api.sigs.k8s.io/concepts/security-model
 
+
+## ARM Image Support
+
+Use buildx for image building to support both amd64 and arm64:
+```
+docker buildx build -f Dockerfile -t registry/admission-server:tag . --platform linux/amd64,linux/arm64
+```
```
Comment (robscott) [inline README.md:77]: I don't think this is needed in README if we update `hack/build-and-push.sh` to use this directly

Comment (wilsonwu) [reply README.md:77]: Yes, I think so.

**Comment by wilsonwu** [comment]
> Hey @wilsonwu, since #1402 has gone stale, I think we can use yours. Do you mind updating this PR to include `hack/build-and-push.sh` updates as well?

No problem, I can update the script for docker image build ASAP.

File: hack/build-and-push.sh [hunk 2]
```diff
@@ -64,10 +64,9 @@ fi
...
-  			-t ${REGISTRY}/admission-server:${GIT_TAG} .
-docker push ${REGISTRY}/admission-server:${GIT_TAG}
+docker buildx build --build-arg COMMIT=${COMMIT} --build-arg TAG=${BINARY_TAG} \
+  			-t ${REGISTRY}/admission-server:${GIT_TAG} . --platform linux/amd64,linux/arm64 --push
 
 # Then, we add an extra version tag - either :latest or semver.
-docker tag ${REGISTRY}/admission-server:${GIT_TAG} ${REGISTRY}/admission-server:${VERSION_TAG}
-docker push ${REGISTRY}/admission-server:${VERSION_TAG}
+docker buildx build --build-arg COMMIT=${COMMIT} --build-arg TAG=${BINARY_TAG} \
+  			-t ${REGISTRY}/admission-server:${VERSION_TAG} . --platform linux/amd64,linux/arm64 --push
```
Comment (shaneutt) [inline hack/build-and-push.sh:72]: If I'm not mistaken, we shouldn't have needed to change the tag line at all now that we're building a multi-arch image? Am I mistaken?

Comment (howardjohn) [reply hack/build-and-push.sh:72]: `buildx build` can do build+push multiple tags in one command.

This should be one `docker` exec with two `-t` flags passed

File: hack/build-and-push.sh [hunk 3]
```diff
@@ -64,10 +64,9 @@ fi
...
-  			-t ${REGISTRY}/admission-server:${GIT_TAG} .
-docker push ${REGISTRY}/admission-server:${GIT_TAG}
-
-# Then, we add an extra version tag - either :latest or semver.
-docker tag ${REGISTRY}/admission-server:${GIT_TAG} ${REGISTRY}/admission-server:${VERSION_TAG}
-docker push ${REGISTRY}/admission-server:${VERSION_TAG}
+# And, we add an extra version tag - either :latest or semver.
+# The buildx integrate build and push in one line.
+docker buildx build --build-arg COMMIT=${COMMIT} --build-arg TAG=${BINARY_TAG} \
+  			-t ${REGISTRY}/admission-server:${GIT_TAG} \
```
Comment (howardjohn) [inline hack/build-and-push.sh:70]: very nit: indent is different

**Review by howardjohn — APPROVED** [review]

**Review by keithmattix — APPROVED** [review]

**Review by shaneutt — APPROVED** [review]
/lgtm, thank you!

**Pull request merged**